File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/AboutUsFragment.java
package org.mifos.mobilebanking.ui.fragments;

/*
~This project is licensed under the open source MPL V2.
~See https://github.com/openMF/self-service-app/blob/master/LICENSE.md
*/

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.google.android.gms.oss.licenses.OssLicensesMenuActivity;

import org.mifos.mobilebanking.BuildConfig;
import org.mifos.mobilebanking.R;
import org.mifos.mobilebanking.ui.activities.PrivacyPolicyActivity;
import org.mifos.mobilebanking.ui.fragments.base.BaseFragment;

import java.util.Calendar;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;


public class AboutUsFragment extends BaseFragment {

    @BindView(R.id.tv_app_version)
    TextView tvAppVersion;

    @BindView(R.id.tv_copy_right)
    TextView tvCopyRight;

    View rootView;

    public static AboutUsFragment newInstance() {
        AboutUsFragment fragment = new AboutUsFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        rootView = inflater.inflate(R.layout.fragment_about_us, container, false);
        ButterKnife.bind(this, rootView);
        setToolbarTitle(getString(R.string.about_us));

        tvAppVersion.setText(getString(R.string.app_version, BuildConfig.VERSION_NAME));

        tvCopyRight.setText(getString(R.string.copy_right_mifos,
                String.valueOf(Calendar.getInstance().get(Calendar.YEAR))));

        return rootView;
    }

    @OnClick(R.id.tv_licenses)
    void showOpenSourceLicenses() {
        startActivity(new Intent(getActivity(), OssLicensesMenuActivity.class));
    }

    @OnClick(R.id.tv_privacy_policy)
    void showPrivacyPolicy() {
        startActivity(new Intent(getActivity(), PrivacyPolicyActivity.class));
    }
}

File: app/src/main/java/org/mifos/mobile/ui/about/AboutUsFragment.kt
package org.mifos.mobile.ui.about

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.google.android.gms.oss.licenses.OssLicensesMenuActivity
import dagger.hilt.android.AndroidEntryPoint
import org.mifos.mobile.ui.activities.PrivacyPolicyActivity
import org.mifos.mobile.ui.fragments.base.BaseFragment
import org.mifos.mobile.core.common.Constants.LICENSE_LINK
import org.mifos.mobile.core.common.Constants.SOURCE_CODE_LINK
import org.mifos.mobile.core.common.Constants.WEBSITE_LINK
import org.mifos.mobile.core.model.enums.AboutUsListItemId
import org.mifos.mobile.core.ui.component.mifosComposeView
import org.mifos.mobile.feature.about.ui.AboutUsScreen


@AndroidEntryPoint
class AboutUsFragment : BaseFragment() {

    @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View {
        return mifosComposeView(requireContext()) {
            AboutUsScreen(
                navigateToItem = {
                    navigateToItem(it.itemId)
                }
            )
        }
    }

    private fun navigateToItem(aboutUsItem: AboutUsListItemId) {
        when(aboutUsItem) {
            AboutUsListItemId.OFFICE_WEBSITE -> {
                startActivity(WEBSITE_LINK)
            }

            AboutUsListItemId.LICENSES -> {
                startActivity(LICENSE_LINK)
            }

            AboutUsListItemId.PRIVACY_POLICY -> {
                startActivity(PrivacyPolicyActivity::class.java)
            }

            AboutUsListItemId.SOURCE_CODE -> {
                startActivity(SOURCE_CODE_LINK)
            }

            AboutUsListItemId.LICENSES_STRING_WITH_VALUE -> {
                startActivity(OssLicensesMenuActivity::class.java)
            }

            AboutUsListItemId.APP_VERSION_TEXT -> Unit
        }
    }

    fun startActivity(uri: String) {
        context?.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(uri)))
    }

    fun <T : Activity> startActivity(clazz: Class<T>) {
        context?.startActivity(Intent(context, clazz))
    }

    companion object {
        fun newInstance(): AboutUsFragment {
            return AboutUsFragment()
        }
    }
}
